dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java

```diff
@@ -1084,4 +1085,150 @@ private ContentType createContentTypeWithCatAndTextField(final Category parentCa
+     * Expected result: false
+     */
+    @Test
+    public void test_isParent_givenGrandChildCategoryThatNoBelongsToParentCategory_returnFalse()
+            throws DotSecurityException, DotDataException {
+        List<Category> categoriesToDelete = Lists.newArrayList();
+        final CategoryAPI categoryAPI = APILocator.getCategoryAPI();
+        try {
+            //Create Parent Category.
+            final Category parentCategory = new CategoryDataGen()
+                    .setCategoryName("CT-Category-Parent")
+                    .setKey("parent")
+                    .setCategoryVelocityVarName("parent")
+                    .nextPersisted();
+
+            categoriesToDelete.add(parentCategory);
+
+            //Create First Child Category.
+            final Category childCategoryA = new CategoryDataGen()
+                    .setCategoryName("CT-Category-A")
+                    .setKey("categoryA")
+                    .setCategoryVelocityVarName("categoryA")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA);
+
+            //Second Level Category.
+            Category childCategoryA_1 = new CategoryDataGen()
+                    .setCategoryName("CT-Category-A-1")
+                    .setKey("categoryA-1")
+                    .setCategoryVelocityVarName("categoryA-1")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA_1);
+
+            categoryAPI.save(parentCategory, childCategoryA, user, false);
+            categoryAPI.save(childCategoryA, childCategoryA_1, user, false);
+
+            //Create Parent Category 2.
+            final Category parentCategory2 = new CategoryDataGen()
+                    .setCategoryName("CT-Category-Parent2")
+                    .setKey("parent2")
+                    .setCategoryVelocityVarName("parent2")
+                    .nextPersisted();
+
+            categoriesToDelete.add(parentCategory2);
+
+            //Create First Child Category 2.
+            final Category childCategoryA2 = new CategoryDataGen()
+                    .setCategoryName("CT-Category-A2")
+                    .setKey("categoryA2")
+                    .setCategoryVelocityVarName("categoryA2")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA2);
+
+            //Second Level Category 2.
+            Category childCategoryA_2 = new CategoryDataGen()
+                    .setCategoryName("CT-Category-A-2")
+                    .setKey("categoryA-2")
+                    .setCategoryVelocityVarName("categoryA-2")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA_2);
+
+            categoryAPI.save(parentCategory2, childCategoryA2, user, false);
+            categoryAPI.save(childCategoryA2, childCategoryA_2, user, false);
+
+            Assert.assertFalse(categoryAPI.isParent(childCategoryA_2,parentCategory,user,true));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertFalse' due to existing static import 'org.junit.Assert.assertFalse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190817)